lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php

```diff
@@ -466,6 +466,20 @@ private function parseColumnCollationFromSQL(string $column, string $sql) : ?str
         return $match[1];
     }
 
+    private function parseTableCommentFromSQL(string $table, string $sql) : ?string
+    {
+        $pattern = '{[\s]*CREATE TABLE(?:\W"' . preg_quote($this->_platform->quoteSingleIdentifier($table)) . '"\W|\W' . preg_quote($table)
+            . '\W)(?:\(.*?\)|[^,(])*?,?((?:(?!\n))(?:\s*--[^\n]*\n?)+)}i';
```

Comment:
This pattern is quite long… maybe we should use the x modifier here.

>  It is also possible to use bracket style delimiters where the opening and closing brackets are the starting and ending delimiter, respectively. (), {}, [] and <> are all valid bracket style delimiter pairs. 

TIL :bulb: , this is nice!

Reply:
Here is how it could look:
```
        $pattern = '{[\s]*CREATE TABLE(?:\W"' . preg_quote($this->_platform->quoteSingleIdentifier($table)) . '"\W|\W' . preg_quote($table). '\W)
(
    ?:\(.*?\)
    |
    [^,(]
)
*?,?(
    (?:(?!\n)) # comment explaining what this is
    (?:\s*--[^\n]*\n?)+ # other comment, etc.
)}ix';
```

Reply:
Excellent idea. Actually, spending some time commenting the regular expression allowed me to spot some useless parts. Done in https://github.com/doctrine/dbal/pull/3512/commits/3e381ef151a78d63c5aa7624f4e0a00e24cb8341

Reply:
Lovely!